docs/en/sql_reference/statements/grant.md

```diff
@@ -0,0 +1,376 @@
+---
+toc_priority: 39
+toc_title: GRANT
+---
+
+
+# GRANT
+
+- Grants [privileges](#grant-privileges) to ClickHouse user accounts or roles.
+- Assigns roles to user accounts or to another roles.
+
+To revoke privileges, use the [REVOKE](revoke.md) statement. Also you can list granted privileges by the [SHOW GRANTS](show.md#show-grants-statement) statement.
+
+## Granting Privilege Syntax {#grant-privigele-syntax}
+
+```sql
+GRANT [ON CLUSTER] privilege[(column_name [,...])] [,...] ON {db.table|db.*|*.*|table|*} TO {user | role | CURRENT_USER} [,...] [WITH GRANT OPTION]
+```
+- `privilege` — Type of privilege.
+- `role` — ClickHouse user role.
+- `user` — ClickHouse user account.
+
+The `WITH GRANT OPTION` clause sets [GRANT OPTION](#grant-option-privilege) privilege for `user` or `role`.
+
+## Assigning Role Syntax {#assign-role-syntax}
+
+```sql
+GRANT [ON CLUSTER] role [,...] TO {user | another_role | CURRENT_USER} [,...] [WITH ADMIN OPTION]
+```
+
+- `role` — ClickHouse user role.
+- `user` — ClickHouse user account.
+
+The `WITH ADMIN OPTION` clause sets [ADMIN OPTION](#admin-option-privilege) privilege for `user` or `role`.
+
+## Usage {#grant-usage}
+
+To use `GRANT`, your account must have the `GRANT OPTION` privilege. You can grant privileges only inside the scope of your account privileges.
+
+For example, administrator has granted privileges to the `john` account by the query:
+
+```sql
+GRANT SELECT(x,y) ON db.table TO john WITH GRANT OPTION
+```
+
+It means that `john` has the permission to perform:
+
+- `SELECT x,y FROM db.table`.
+- `SELECT x FROM db.table`.
+- `SELECT y FROM db.table`.
+
+`john` can't perform `SELECT z FROM db.table`. The `SELECT * FROM db.table` also is not available. ClickHouse doesn't return any data, even `x` and `y`.
+
+Also `john` has the `GRANT OPTION` privilege, so it can grant other users with privileges of the same or the smaller scope.
+
+Specifying privileges you can use asterisk (`*`) instead of a table or a database name. For example, the `GRANT SELECT ON db.* TO john` query allows `john` to perform the `SELECT` query over all the tables in `db` database. Also, you can omit database name. In this case privileges are granted for current database.
+
+You can grant multiple privileges to multiple accounts in one query. The query `GRANT SELECT, INSERT ON *.* TO john, petya` allows accounts `john` and `petya` to perform the `INSERT` and `SELECT` queries over all the tables in all the databases on the server.
```

Comment:
Please add here the explanation about current database:

GRANT SELECT ON * TO john - gives the privilege on any tables in the current database
GRANT SELECT ON mytable TO john - gives the privilege on the table `mytable` in the current database